packages/tools/client-debug-view/src/components/data-object-views/SharedCounterView.tsx

```diff
@@ -37,12 +54,65 @@ export function SharedCounterView(props: SharedCounterViewProps): React.ReactEle
 		};
 	}, [sharedCounter, setValue]);
 
+
+    function decrementCounter(): void {
+        sharedCounter.increment(-deltaValue);
+	}
+
+    function incrementCounter(): void {
+		sharedCounter.increment(deltaValue);
+	}
+
+    const inputSetValue = (e: React.ChangeEvent<HTMLInputElement>): void => {
+        numberFormatCheck(e.target.value);
+    }
+
+    const numberFormatCheck = (val: string): void => {
+        const num = Number.parseInt(val, 10) || value;
+
+        if (!Number.isFinite(num) || num[0] === '-') {
+            return
+        }
+
+        setDeltaValue(num);
+    }
+
 	return (
 		<Stack>
 			<StackItem>
 				<b>SharedCounter</b>
 			</StackItem>
-			<StackItem>Value: {value}</StackItem>
+			<StackItem>Value: {value} </StackItem>
+            <StackItem>
+                <TooltipHost
+					content="Decrement counter by 1 (min 0)."
```

Comment:
We'll need to update the tooltip text here, since these buttons don't increment / decrement by 1